src/Axe/Managers/CompileManager.cs

```diff
@@ -0,0 +1,268 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Axe.Models;
+using Axe.ViewModels.CompilerVm;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis;
+using Microsoft.EntityFrameworkCore;
+using System.IO;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using System.Reflection;
+using System.Collections;
+using Microsoft.CodeAnalysis.Emit;
+using System.Runtime.Loader;
+
+namespace Axe.Managers
+{
+    /// <summary>
+    /// Class implements operations which can be performed with <see cref="CodeBlock"/> entities
+    /// </summary>
+    public class CompileManager : ManagerBase, ICompileManager
+    {
+        public CompileManager(AxeDbContext context) : base(context)
+        {
+        }
+        
+        public async Task<List<CodeBlockVm>> GetCodeBlocks()
+        {
+            List<CodeBlockVm> list = await context.CodeBlock.Select(cb => new CodeBlockVm
+            {
+                Id = cb.Id,
+                Task = cb.Task
+            }).ToListAsync();
+            return list;
+        }
+
+        public async Task<CodeBlockVm> GetById(int id)
+        {
+            try
+            {
+                CodeBlock codeBlock = await this.context.CodeBlock.FirstAsync(cb => cb.Id == id);
+                CodeBlockVm model = new CodeBlockVm()
+                {
+                    Id = codeBlock.Id,
+                    Task = codeBlock.Task
+                };
+                model.SourceCode = CSharpSyntaxTree.ParseText(model.SourceCode).GetRoot().NormalizeWhitespace().ToFullString();
+                return model;
+            }
+            catch (Exception exception)
+            {
+                throw exception;
+            }
+        }
+
+        public Tuple<CodeBlockResult, string[]> Solve(CodeBlockVm model)
+        {
+            try
+            {
+                CodeBlock codeBlock = context.CodeBlock
+                    .Include(cb => cb.TestCases)
+                    .First(cb => cb.Id == model.Id);
+
+                string assemblyName = "AxeCompilerAssembly" + Path.GetRandomFileName();
+                // Random file name is necessary because .NETCore doesn't allow to unload assembly
```

Comment:
try load/unload assembly in a separate Application Domain

Reply:
.NET Core doesn't support Application Domain: https://github.com/dotnet/standard/blob/master/docs/faq.md#is-appdomain-part-of-net-standard